model_analyzer/model_manager.py

```diff
@@ -74,9 +74,11 @@ def run_models(self, models):
         rcg = RunConfigGenerator(config=self._config,
                                  models=models,
                                  client=self._client)
+
+        run_config = rcg.next_config()
         while not rcg.is_done() and not self._state_manager.exiting():
-            run_config = rcg.next_config()
-            measurement = self._metrics_manager.execute_run_config(run_config)
+            measurement = self._metrics_manager.execute_run_config(
+                next(run_config))
```

Comment:
how is `next()` `StopIteration` exception ([link](https://docs.python.org/3/library/functions.html#next)) avoided here? is it tied to `rcg.is_done()`?